internal/compose/compose.go

```diff
@@ -316,12 +321,19 @@ func (p *Project) WaitForHealthy(opts CommandOptions) error {
 			healthy = false
 		}
 
+		// end loop before timeout if healthy
 		if healthy {
 			break
 		}
 
-		time.Sleep(time.Second)
+		// NOTE: using sleep does not guarantee interval but it's ok for this use case
+		time.Sleep(interval)
 	}
+
+	if !healthy {
```

Comment:
the bug is obvious, I misinterpreted the logic. `healthy` must be reinitialized at each loop, otherwise once it become `false` the logic does not change it's value anymore. I tried to use it to detect the timeout, but I stumbled upon the bug. The current implementation is clearer and uses times instead of a side effect on the `healthy` value.